### ISSUE TITLE
Form fields in web pages in project templates appear squished together

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/aspnetcore#40887]
**Affected versions**: .NET 6 and .NET 7

Seems as part of the migration to Bootstrap 5.x the styling in our templates has resulted in form fields being squashed together in a very unattractive way, e.g.:

<img width="299" alt="image" src="https://user-images.githubusercontent.com/249088/160214245-9fe25d94-cb22-4538-9971-23ab06c612ff.png">

Compare this to example form on the [Bootstrap docs](https://getbootstrap.com/docs/5.1/forms/overview/):

<img width="313" alt="image" src="https://user-images.githubusercontent.com/249088/160214261-fe711652-5b10-42d0-8b38-296451a9b513.png">

Both screenshots were taken in Edge latest running on Windows 11 running full screen at 4K at 150% scaling. The templates used were latest from the `main` branch but this reproduces in .NET 6 too.

The effect is exacerbated when an input text field has focus as its border highlight is clipped by the next element in the form, e.g. the button is overlapping the focus border effect of the input field:'

<img width="291" alt="image" src="https://user-images.githubusercontent.com/249088/160214606-6e7c8696-c8ce-4cf4-898f-c4623c6e07fc.png">

**Comment by javiercn** [comment]
@DamianEdwards Yeah, I remember discussing this with @HaoK at the time.

I believe this was actually how Bootstrap 5 looks like (at least in some config). @HaoK do you remember the details?

I've looked in the Bootstrap docs and at least a bunch of the form fields look like this, although there are others that present a more "classic" style. I agree its not ideal and we should probably consider changing them.

I've filed a separate issue to discuss the changes/updates for the templates in .NET 7.0 [here](https://github.com/dotnet/aspnetcore/issues/40935) since there seems to be no issue for that.

**Comment by HaoK** [comment]
Yeah this was just their default look, definitely not something we went out of our way to choose.  I think the only things we tweaked was some of the colors to pass the contrast rules for accessibility

**Comment by HaoK** [comment]
Their login example doesn't look that diff
![image](https://user-images.githubusercontent.com/6537861/160646953-eea0ca92-f594-4f8e-b890-303bef325174.png)

**Comment by DamianEdwards** [comment]
There are subtle differences that (IMO) make a big difference. The stacked input fields example above from their sample removes the rounded corners from the sides where the fields touch each other so it looks far more intentional. I'm not super fussed which style we go for as long as we implement it to match.

**Comment by TanvirArjel** [comment]
@HaoK This is how  bootstrap 5.x form floating looks like:

![image](https://user-images.githubusercontent.com/14342773/170095649-a049f50c-9df9-4631-aae6-3564fbbfef30.png)

Links: https://getbootstrap.com/docs/5.2/forms/floating-labels/#example

**Comment by HaoK** [comment]
@blowdart okay with adding the placeholder values like so?  The mb-3 change seems easy enough

**Comment by blowdart** [comment]
Sure :)

**Comment by HaoK** [comment]
@deepchoudhery I'm not sure how we are tracking scaffolding updates for 7.0, do you generally do a bulk update, or do you want me to file a one off PR in scaffolding?

**Comment by deepchoudhery** [comment]
@HaoK I have been doing bulk updates, would prefer that, or in 2-3 smaller bulk ones if you prefer, instead of a file at a time. Thanks.

**Comment by HaoK** [comment]
Ok cool, so you will do a bulk update for 7.0 at some future date?

**Comment by deepchoudhery** [comment]
Yes, I will be sifting through identity updates and making changes for next preview. I prefer to do it in the later previews so it's more finalized for you guys by then.
I'll tag you in the PR so expect one within 2-3 weeks.

Thanks,
Deep.
